File: cypress-js/cypress/support/pages/CapturePage.js
import BasePage from './BasePage';

const CAMERA_CARD_TEXT = 'Tomar Foto';
const GALLERY_CARD_TEXT = 'Galería';
const EXPLAINABILITY_TEXT = 'Explicabilidad';
const FORMATS_INFO_TEXT = 'Formatos soportados';
const PAGE_HEADER_TEXT = 'Analizar Imagen';

/**
 * Page Object for the Capture (image-upload) screen. Constructing this
 * object waits until the "Tomar Foto" card is visible.
 */
export default class CapturePage extends BasePage {
  constructor() {
    super();
    this.byPartialText(CAMERA_CARD_TEXT).should('be.visible');
  }

  // ── Assertions ────────────────────────────────────────────────────────────

  assertCameraOptionVisible() {
    this.byPartialText(CAMERA_CARD_TEXT).should('be.visible');
    return this;
  }

  assertGalleryOptionVisible() {
    this.byPartialText(GALLERY_CARD_TEXT).should('be.visible');
    return this;
  }

  assertExplainabilityVisible() {
    this.byPartialText(EXPLAINABILITY_TEXT).should('be.visible');
    return this;
  }

  assertFormatsInfoVisible() {
    this.byPartialText(FORMATS_INFO_TEXT).should('be.visible');
    return this;
  }

  assertPageHeaderVisible() {
    this.byPartialText(PAGE_HEADER_TEXT).should('be.visible');
    return this;
  }
}

File: playwright-csharp/Pages/CapturePage.cs
using Microsoft.Playwright;

namespace PlaywrightStClimanuvem.Pages;

/// <summary>
/// Page Object for the Capture (image-upload) screen. <see cref="CreateAsync"/>
/// waits until the "Tomar Foto" card is visible. Mirrors selenium-java's
/// <c>CapturePage</c>.
/// </summary>
public sealed class CapturePage : BasePage
{
    private const string CameraCardText = "Tomar Foto";
    private const string GalleryCardText = "Galería";
    private const string ExplainabilityText = "Explicabilidad";
    private const string FormatsInfoText = "Formatos soportados";
    private const string PageHeaderText = "Analizar Imagen";

    private CapturePage(IPage page) : base(page)
    {
    }

    public static async Task<CapturePage> CreateAsync(IPage page)
    {
        var capturePage = new CapturePage(page);
        await WaitForAsync(capturePage.CameraCard());
        return capturePage;
    }

    // ── Queries ──────────────────────────────────────────────────────────

    public Task<bool> IsCameraOptionVisibleAsync() => IsPresentAsync(CameraCard());
    public Task<bool> IsGalleryOptionVisibleAsync() => IsPresentAsync(ByPartialText(GalleryCardText));
    public Task<bool> IsExplainabilityVisibleAsync() => IsPresentAsync(ByPartialText(ExplainabilityText));
    public Task<bool> IsFormatsInfoVisibleAsync() => IsPresentAsync(ByPartialText(FormatsInfoText));
    public Task<bool> IsPageHeaderVisibleAsync() => IsPresentAsync(ByPartialText(PageHeaderText));

    private ILocator CameraCard() => ByPartialText(CameraCardText);
}
